src/navsections/sectionBlueprintMiddleware/changes.ts

```diff
@@ -0,0 +1,19 @@
+import {shallowEqual} from 'react-redux';
+
+import {NavigatorInstanceState} from '@models/navigator';
+
+import {pickPartialRecord} from './utils';
+
+export const hasNavigatorInstanceStateChanged = (
+  navigatorState: NavigatorInstanceState,
+  newNavigatorInstanceState: NavigatorInstanceState
+) => {
+  const {itemInstanceMap, sectionInstanceMap} = newNavigatorInstanceState;
+  return (
+    !shallowEqual(pickPartialRecord(navigatorState.itemInstanceMap, Object.keys(itemInstanceMap)), itemInstanceMap) ||
```

Comment:
Maybe here it would be better to use lodash `isEqual` as shallowEqual didn't work as expected in some cases